src/ScottLogic.Internal.Training/ScottLogic.Internal.Training.Api/Controllers/OrdersController.cs

```diff
@@ -72,7 +73,16 @@ public IActionResult Get(int userAccountNumber)
         [Route("buy")]
         public IActionResult Buy([FromBody] Order currentOrder)
         {
-            if (currentOrder.Action == OrderType.Buy)
+            // Get the user account number from the access token
+            var userIdentity = this.User.Identity as ClaimsIdentity;
+            var userAccount = 0;
+            if (userIdentity != null && userIdentity.HasClaim(c => c.Type == "AccountNumber"))
```

Comment:
What happens if my token doesn't have an account number, and the order I pass in has an account number of 0?

0 is the default value for an integer and using default values in checks like this is never a good idea. Look into nullable types, using those should remove any risk of us accidentally setting the default value to a valid acount number.